Xamarin.PropertyEditing.Windows/CategoryExpander.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Media;
+using Xamarin.PropertyEditing.ViewModels;
+
+namespace Xamarin.PropertyEditing.Windows
+{
+	internal class CategoryExpander
+		: Expander
+	{
+		public CategoryExpander ()
+		{
+			DataContextChanged += OnDataContextChanged;
+		}
+
+		public static readonly DependencyProperty IsFilteredProperty = DependencyProperty.Register (
+			"IsFiltered", typeof(bool), typeof(CategoryExpander), new PropertyMetadata (false, (o, args) => ((CategoryExpander)o).OnIsFilteredChanged()));
+
+		public bool IsFiltered
+		{
+			get { return (bool) GetValue (IsFilteredProperty); }
+			set { SetValue (IsFilteredProperty, value); }
+		}
+
+		public override void OnApplyTemplate ()
+		{
+			base.OnApplyTemplate ();
+			UpdateValue();
+		}
+
+		protected override void OnExpanded ()
+		{
+			base.OnExpanded ();
+			SetExpanded (true);
+		}
+
+		protected override void OnCollapsed ()
+		{
+			base.OnCollapsed ();
+			SetExpanded (false);
+		}
+
+		private bool fromFilter;
+		private PanelViewModel panelVm;
+
+		private void UpdateValue ()
+		{
+			if (this.panelVm == null)
+				return;
+
+			SetCurrentValue (IsExpandedProperty, IsFiltered || this.panelVm.GetIsExpanded (Header as string));
+		}
+
+		private void OnIsFilteredChanged ()
+		{
+			this.fromFilter = true;
+			UpdateValue ();
+			this.fromFilter = false;
+		}
+
+		private void UpdateViewModel ()
+		{
+			FrameworkElement element = this;
+			while (element != null && !(element is ItemsControl)) {
```

Comment:
FYI, it happens I just added a helper for that: `XamlHelper.FindAncestor<T>(this DependencyObject control)`

Reply:
Which is in an unmerged PR...